src/modules/service-catalog/components/useItemFormFields.tsx

```diff
@@ -0,0 +1,108 @@
+import { useCallback, useEffect, useState } from "react";
+import type { ServiceCatalogItem } from "../data-types/ServiceCatalogItem";
+import type { TicketField } from "../data-types/TicketField";
+import type { Field } from "../../ticket-fields/data-types/Field";
+import { getCustomObjectKey } from "../../ticket-fields/fields/LookupField";
+
+const formatField = (field: TicketField): Field => {
+  const {
+    id,
+    type,
+    description,
+    title_in_portal,
+    custom_field_options,
+    required_in_portal,
+    relationship_target_type,
+  } = field;
+  return {
+    id,
+    type,
+    name: `custom_fields_${id}`,
+    description,
+    label: title_in_portal,
+    options: custom_field_options,
+    required: required_in_portal,
+    relationship_target_type,
+    error: null,
+  };
+};
+
+const isAssociatedLookupField = (field: TicketField) => {
+  const customObjectKey = getCustomObjectKey(
+    field.relationship_target_type as string
+  );
+  return customObjectKey === "service_catalog_item";
+};
+
+const fetchTicketFields = async (form_id: string): Promise<Field[]> => {
+  try {
+    const [formResponse, fieldsResponse] = await Promise.all([
+      fetch(`/api/v2/ticket_forms/${form_id}`),
+      fetch(`/api/v2/ticket_fields`),
+    ]);
+
+    if (!formResponse.ok) {
+      throw new Error("Error fetching form data");
+    }
+    if (!fieldsResponse.ok) {
+      throw new Error("Error fetching fields data");
+    }
+
+    const formData = await formResponse.json();
+    const fieldsData = await fieldsResponse.json();
+
+    const ids = formData.ticket_form.ticket_field_ids;
+    const ticketFieldsData = fieldsData.ticket_fields.filter(
+      (field: TicketField) => ids.includes(field.id)
+    );
+    const requestFields = ticketFieldsData
+      .map((ticketField: TicketField) => {
+        if (
+          ticketField.type === "lookup" &&
+          isAssociatedLookupField(ticketField)
+        ) {
+          return null;
+        } else {
+          return formatField(ticketField);
+        }
+      })
+      .filter(Boolean);
+    return requestFields;
```

Comment:
We need to take into account a couple of more things:
- the `ticket_field_ids` are ordered and we need to respect the same order when we show them. With this implementation, we use the order returned by the ticket fields API
- we also need to show only fields that have `editable_in_portal: true`. In theory, we should not have fields with this property set to false, but better add this check for safety
- we are returning fields that we don't show later in the form (subject, description, assignee, group, ...). It is not a big issue since we are not showing them, but better to not keep them in the state as well. Most of them will be filtered out if we keep only `editable_in_portal: true`, at that point probably we will need to filter out just subject and description

Reply:
Ok changed code to respect the order from the form and added `base_locale` to the API